src/testdrive/src/parser.rs

```diff
@@ -476,6 +478,26 @@ impl<'a> Iterator for BuiltinReader<'a> {
                         ),
                     }));
                 }
+            } else if c == '\'' {
+                if nesting.is_empty() {
+                    nesting.push(c);
+                } else {
+                    if let Some(&'\'') = nesting.last() {
+                        nesting.pop();
+                    } else {
+                        return Some(Err(InputError {
+                            pos: self.pos + i,
+                            msg: "command argument has unterminated single quote".to_owned(),
+                        }));
+                    }
+                }
+                continue;
+            } else if c == '\"' {
+                if let Some(&'\"') = nesting.last() {
+                    nesting.pop();
+                } else {
+                    nesting.push(c);
+                }
```

Comment:
This code is a bit confusing to me! Strings don't really nest the way that `{` and `}` and `[` and `]` do, since you can't tell whether a given quote is meant to start a new nesting or pop out of the old one. I'd expect that you just parse until you see the closing quote, possibly ignoring backslash escaped quotes. But see below for a way we might be able to avoid dealing with the parser. 

(To be clear, would be happy to learn for testdrive to accept quoted argument values, but I think it would need some tweaks.)

Reply:
So this looks to me like it’ll still have some nesting funniness—can you either back out, tweak, or file a follow up task?